nlohmann_json.natvis

```diff
@@ -20,13 +24,253 @@
         </Expand>
     </Type>
 
-    <!--    skip the pair first/second members in the treeview while traversing a map.
-            Only works in VS 2015 Update 2 and beyond using the new visualization -->
+    <!-- Skip the pair first/second members in the treeview while traversing a map.
+         Only works in VS 2015 Update 2 and beyond using the new visualization -->
     <Type Name="std::pair&lt;*, nlohmann::basic_json&lt;*&gt;&gt;" IncludeView="MapHelper">
         <DisplayString>{second}</DisplayString>
         <Expand>
             <ExpandedItem>second</ExpandedItem>
         </Expand>
     </Type>
 
+    <!-- Namespace nlohmann::json_abi -->
+    <Type Name="nlohmann::json_abi::basic_json&lt;*&gt;">
```

Comment:
Is it possible to use wildcards for the namespace?  `nlohmann::*::basic_json...`?  If so, could we also hardcode the values for `nlohmann::json_abi::detail::value_t::*`?  This would allow a single natvis file that would work for any version?

Otherwise, would it make sense to include the version number in the filename so that users could have multiple version files installed?

Reply:
Wildcards are only supported for template parameters according to the documentation.

Users will have to comment on whether versioning the Natvis file would be helpful. (ping @jheydebrand)

Reply:
>Wildcards are only supported for template parameters according to the documentation.

I saw documentation that said that they worked for templates, but it didn't specifically say that it was only used there, so I wasn't sure if it was limited to templates or just poorly documented.

Reply:
> The fully qualified name of each Type element is specified in its Name attribute.

and

> ### Templated classes
> The Name attribute of the Type element accepts an asterisk * as a wildcard character that can be used for templated class names.
>
> In the following example, the same visualization is used whether the object is a CAtlArray<int> or a CAtlArray<float>. If there's a specific visualization entry for a CAtlArray<float>, then it takes precedence over the generic one.

The very specific wording suggests to me it only works for templates. :man_shrugging:

Reply:
Confirmed that it only works for templates. :(

Reply:
Thanks for the swift responses.

Versioning the natvis file would not provide any benefit for me.
When consuming the library via its CMake integration the correct / matching natvis file will be included anyway and I have no desire to copy the file to some system-wide directory (which is supported by Visual Studio and might have been the use-case for versioned natvis files you were thinking about?).